dashboard/lib/pd/survey_pipeline/survey_rollup_decorator.rb

```diff
@@ -0,0 +1,135 @@
+module Pd::SurveyPipeline
+  class SurveyRollupDecorator < SurveyPipelineWorker
+    def self.process_data(context)
+      check_required_input_keys self::INPUT_KEYS, context
+
+      results = decorate(*context.values_at(*self::INPUT_KEYS))
+
+      self::OUTPUT_KEYS.each do |key|
+        context[key] ||= {}
+        context[key].deep_merge! results[key]
+      end
+
+      context
+    end
+
+    def self.decorate(summaries, facilitator_id, current_workshop_id, related_workshop_ids,
+      submissions, submission_type, parsed_questions, question_categories, errors = [])
+      result = {
+        facilitators: {},
+        current_workshop: current_workshop_id,
+        related_workshops: {},
+        facilitator_response_counts: {this_workshop: {}, all_my_workshops: {}},
+        facilitator_averages: {},
+        errors: errors
+      }
+
+      # Populate result[:facilitators] = {fac_id => fac_name}
+      fac_name = User.find(facilitator_id)&.name || "UserId_#{facilitator_id}"
```

Comment:
optional nit: `facilitator_name` feels more consistent, ruby-ish, and a teeny bit easier to read

Reply:
I forgot to clean up these short variable names, doing it now.